### PR TITLE
Correlate sent errors with received errors

### LINKED ISSUE #2045: Error identifiers are not sufficiently unique for debugging a distributed system
**What is the Problem Being Solved?**

Following the addition of #2037, we have a means of correlating the causal graph of error locally, but the error identifiers are not robust against collisions across vats.

**Description of the Design**

We need to thread an identifier for vats to the error cause notes emitted by the marshal layer such that they are sufficiently unique that they can be correlated with reasonable confidence that they will not collide across a distributed system.

In relatively insular systems, it may be sufficient for the causal trace to use the memorable name of a vat for this unique identifier. For a distributed system, this will likely need to be an arbitrary human readable name like “Alice M. X-Ray” chosen consistently based on some number of parameters including the public key of a node or an endo, for example.

**Security Considerations**

Distributed debuggers stand in a position of extraordinary privilege, being privy to the stacks of every node on the system that elects to emit them. Such systems might elide platform frames from the stacks they emit.

**Test Plan**

To test this will require a scaffold that orchestrates multiple vats and collates the causal error graph.

### LINKED ISSUE #2076: Deploy script error messages are not as helpful as they should be
**Describe the bug**
Screen Shot 2020-12-10 at 11 19 37 AM

The error stack gives information that is too low level to be helpful, and the things hidden in `details` are not displayed.

**To Reproduce**
1. Check out branch `petname-not-found-error-message` on agoric-sdk
2. Check out branch `petname-not-found-error-message` on https://github.com/Agoric/dapp-loan
3. On agoric-sdk:
`yarn install`
`yarn build`
4. On dapp-loan:
`agoric install`
`agoric start --reset -v`
5. In a separate terminal on dapp-loan:
agoric deploy contract/scripts/setup/startAutoswap.js
agoric deploy contract/scripts/setup/periodAsyncIterable.js
agoric deploy contract/scripts/contractDeveloper/installLoan.js
agoric deploy contract/scripts/lender/startInstance.js

**Expected behavior**
I would like to see the petname revealed and the lines in lib-wallet.js in agoric-sdk and the deploy script in the stack trace

### PR DESCRIPTION
**Pull request by erights**
Initially just a port of #2037 and #2044 from the hackathon branch onto master.

Fixes #2076 

Will not by itself fix #2045 but may make some progress in that direction.

### CODE REVIEW

**Review by katelynsills — APPROVED** [review]
This is a review of the outcome rather than a code review, but as I commented in https://github.com/Agoric/agoric-sdk/issues/2076#issuecomment-744642565 this solves a large part of the bad user experience, which is fantastic!

It doesn't give a line number in the deploy script, however, so I think the user experience could still be improved further. Is that something we could easily fix or is that something that we should create another issue for? Thanks!

**Comment by erights** [comment]
> This is a review of the outcome rather than a code review

Thanks. But I should still wait for a code review before merging.

> It doesn't give a line number in the deploy script,

What line number in the deploy script would we like it to give?

**Comment by katelynsills** [comment]
> Thanks. But I should still wait for a code review before merging.

Ok, I haven't been part of the marshal or error handling layers, so I don't think my code review should be used for merging authorization, but I can provide one if that would be helpful. 

> What line number in the deploy script would we like it to give?

I believe it should be line 97 in `contract/scripts/lender/startInstance.js` that is listed in an error stack:

Screen Shot 2020-12-14 at 11 25 01 AM

**Comment by erights** [comment]
> I believe it should be line 97 in contract/scripts/lender/startInstance.js that is listed in an error stack:

Success! See https://github.com/Agoric/agoric-sdk/issues/2076#issuecomment-745097828

**Comment by erights** [comment]
@kriskowal @katelynsills 
https://github.com/Agoric/agoric-sdk/pull/2082#issuecomment-744657956 is fixed.
Ready for review.

Kate only did an outcome review of the previous state so the "1 approval" doesn't count. I still need a code review before merging.

**Comment by katelynsills** [comment]
Fantastic! I will do my best to review tomorrow unless @kriskowal  gets there first

File: packages/marshal/src/marshal.js [hunk 1]
```diff
@@ -480,7 +480,18 @@ const defaultSlotToValFn = (x, _) => x;
 export function makeMarshal(
   convertValToSlot = defaultValToSlotFn,
   convertSlotToVal = defaultSlotToValFn,
+  // Messages are the medium
+  { marshalName = 'McLuhan' } = {},
```
Comment (katelynsills) [inline packages/marshal/src/marshal.js]: Is this a reference to this? https://www.youtube.com/watch?v=ImaH51F4HBw

**Review by katelynsills — APPROVED** [review]
Did a code review. looks good to me!

File: packages/marshal/src/marshal.js (hunk 1 above)
Comment (erights) [reply packages/marshal/src/marshal.js]: Thanks for reminding me of that. I didn't know about the YouTube vid but altogether yes. Marshal McLuhan was a famous media theorist, with "The Medium is the Message" being his famous tag line.

Although I amused myself for a moment with this obscure cultural reference, when I needed a default marshalName, it does not belong in production code. I just changed the default to `'anon-marshal'`.

Comment (katelynsills) [reply packages/marshal/src/marshal.js]: sounds good!

Comment (kriskowal) [reply packages/marshal/src/marshal.js]: Aside:

My default Marshall is Thurgood.

Etymology of marshal came up the other day. The mar is mare (equestrian) and shal is farrier (as in iron, smith). Something to do with being in charge of lining up horses.

**Pull request merged**